yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -515,7 +515,28 @@ private[spark] class Client(
     logInfo("Setting up container launch context for our AM")
     val appId = newAppResponse.getApplicationId
     val appStagingDir = getAppStagingDir(appId)
-    val localResources = prepareLocalResources(appStagingDir)
+    var localResources: Map[String, LocalResource] = null
+    try {
+      localResources = prepareLocalResources(appStagingDir)
+    } catch {
+      case e: Throwable =>
+        var stagingDirPath: Path = null
```

Comment:
I don't think this variable is needed? `appStagingDir` can be used everywhere except where it needs to be wrapped for `FileSystem`. Can `localResources` remain a `val` and receive the value of the `try` block?